src/librustc_trans/back/lto.rs

```diff
@@ -167,7 +167,12 @@ pub fn run(sess: &session::Session, llmod: ModuleRef,
         llvm::LLVMRustAddAnalysisPasses(tm, pm, llmod);
         llvm::LLVMRustAddPass(pm, "verify\0".as_ptr() as *const _);
 
-        let opt = sess.opts.cg.opt_level.unwrap_or(0) as libc::c_uint;
+        let opt = match sess.opts.optimize {
+            config::No => 0,
```

Comment:
It looks like `-Clto` without optimizations doesn't even perform dead code elimination, which makes LTO rather pointless, IMO.
Can we please make `-Clto` imply `-Copt-level=2`? (unless specified explicitly, of course)